core/src/main/java/org/apache/iceberg/variants/ShreddedObject.java

```diff
@@ -61,7 +61,7 @@ VariantMetadata metadata() {
   }
 
   private Set<String> nameSet() {
-    Set<String> names = Sets.newHashSet(shreddedFields.keySet());
+    Set<String> names = Sets.newTreeSet(shreddedFields.keySet());
```

Comment:
Probably not intended to be a part of this PR

Reply:
Yeah, oops.